hvpy/utils.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Callable
+
+__all__ = ["add_shared_docstring"]
+
+
+def add_shared_docstring(input_class) -> Callable[[Any], Any]:
+    def decorator(func):
+        split_doc = input_class.__doc__.split("{Shared}")
```

Comment:
We want to check if "Shared" is in the documentation string, if it is, do this else just return the func.